### PR TITLE
Python 2.7.13: Fix unicode paths passed to ctypes.CDLL, ctypes.WINDLL, etc.

### LINKED ISSUE #6705: Eurobraille driver does not load anymore with Next version 13776,45cf45d9
With this version, NVDA seems to see the driver, but its code is nomore executed with no error message and nothing special in log file.
the code is simply ignored.
Since it used to work until Next version 13761,fc9ed0a7
There obviously is something in next 13776,45cf45d9
which prevents it from running.
could there be a link with tne fact taht a new checkbox about blinking cursor appeared in the braille settings dialog?
The driver can be found at:
http://www.nvda-fr.org/addons/Erbnvda2014-05-10.nvda-addon

### PR DESCRIPTION
**Pull request by jcsteh**
Monkey patch ctypes._dlopen (used by the constructor for ctypes.CDLL, ctypes.WINDLL, etc.) so that it accepts unicode paths as it did prior to Python 2.7.13.
In Python 2.7.13, it raises an exception if you pass a unicode path. In Python < 2.7.13, it silently converted them to str, treating them as ASCII. Some add-ons (such as the Eurobraille driver) relied on the previous behaviour.
This monkey patch actually gives slightly better results than < 2.7.13 in that it converts paths using mbcs, so it should work with non-ASCII paths as well in most cases. Aside from restoring the previous behaviour, given that we push for unicode wherever possible, I think this is a good thing.
Fixes #6705.

### CODE REVIEW

**Comment by jcsteh** [comment]
I think we should merge this one straight to master once approved, as Python 2.7.13 is already on master.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**
